jwst/photom/photom.py

```diff
@@ -115,7 +115,11 @@ def __init__(self, model, inverse=False, source_type=None, correction_pars=None)
             if model.meta.instrument.grating is not None:
                 self.grating = model.meta.instrument.grating.upper()
             self.order = None
-            if model.meta.get('wcsinfo.spectral_order') is not None:
+            if (
+                model.meta.hasattr('wcsinfo')
+                and model.meta.wcsinfo.hasattr('spectral_order')
+                and model.meta.wcsinfo.spectral_order is not None
+            ):
```

Comment:
I tried that first, but tests fail due to wcsinfo being None.  I didn't look into why that might be.

Reply:
All datamodels should have a `wcsinfo` block, i.e. include that schema, or else they should not even make it to the photometric calibration step in stage 2 pipeline.  So do we have unit tests that are testing for something that will never occur in the wild?

Reply:
Ah, it looks like this is to handle `MultiSlitModel` where the `wcsinfo` block is in

```python
for slit in model.slits:
    self.spectral_order = slit.meta.wcsinfo.spectral_order
```

So this seems like a not great way to initialize the `DataSet` class, because the question becomes how is order determined (if not on `__init__`) for `MultiSlitModel`?  Not the same as for the others.

Anyway, not worth debugging now.